fastapi_websocket_rpc/websocket_rpc_client.py

```diff
@@ -205,12 +333,13 @@ async def reader(self):
         # task was canceled
         except asyncio.CancelledError:
             pass
-        except websockets.exceptions.ConnectionClosed:
-            logger.info("Connection was terminated.")
-            await self.close()
-        except:
-            logger.exception("RPC Reader task failed")
-            raise
+        except Exception as err:
+            if await self.ws.isConnectionClosedException(err):
```

Comment:
Would be better to use exception cases instead of if clauses (more efficient) , this can be achieved by taking the entire function into polymorphism

Reply:
Done! Created BaseConnectionClosedException

Reply:
Sorry for dealyed reposnses, this week is very busy. I've cleared time to review next week.

Reply:
Hi @orweis , maybe if you have just an example, then I could adapt the code? Thanks!

Reply:
Hi, sorry for dropping the ball here, this last week has been extremely busy with end of year around the corner.

I'll do my best to provide you with more guidence later today

Reply:
I do not see how to apply your guidance below (putting exception handling in connect()) here. If we put exception handling inside recv(), then we wouldn't be able to get out of the while loop. Or maybe you want each websocket client handler to implement the entire reader() method? Then there would be a bit of duplicate code. I will try that today.

Reply:
Good point re- recv() / reader() exceptions .
How about having a shared _recv() and have each Class in its recv() call it and catch its exceptions and raise instead a common shared exceptions  (e.g. a joint ConnectionClosed)

Reply:
I have been thinking a bit about it... To my mind, I find it ok to have a reraise, in case of connection closed, because it performance-wise, it only happens once, then the framework close channel, etc.
But anyway, I think about simplicity and modified the code, so that we only expect a "None" as a special return result, in case of connection closed. No need to reraise, no need for a special base exception!

Reply:
https://github.com/antoinetran/fastapi_websocket_rpc/commit/7d899469f23299b7e808729acd0bbfba7a9d0143